From: Vasundhara Volam <vasundhara.volam@emulex.com>
Date: Thu, 17 Jul 2014 16:20:30 +0530
Subject: be2net: avoid SRIOV config for BE2 chip
Patch-mainline: v3.17-rc1
Git-commit: 962bcb750b47ef0c8d28cc217ec22b4e44413565
References: bnc#908322 FATE#317535

As SRIOV is not supported on BE2 chip, avoid calling be_get/set_sriov_config()
for BE2 chip.

Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/emulex/benet/be_main.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -3300,7 +3300,9 @@ static int be_get_config(struct be_adapt
 		if (!status)
 			dev_info(&adapter->pdev->dev,
 				 "Using profile 0x%x\n", profile_id);
+	}
 
+	if (!BE2_chip(adapter) && be_physfn(adapter)) {
 		status = be_get_sriov_config(adapter);
 		if (status)
 			return status;
